integration/spark/src/main/scala/org/apache/spark/sql/execution/command/management/CarbonInsertIntoCommand.scala

```diff
@@ -294,6 +294,13 @@ case class CarbonInsertIntoCommand(databaseNameOp: Option[String],
         }
         throw ex
     }
+    if (internalOptions.contains("isMultipleInserts")) {
+      if (internalOptions.get("isMultipleInserts").get.toBoolean) {
+        // when multiple insert statements are given with single plan (Union),
+        // no need to return segment id as row.
+        return Seq.empty
```

Comment:
here we are returning the segment ID from load command so that the application code integrated with carbondata can take decisions based on that. So if we don't return in this case, we will not give valid info to underlying application and may lead to wrong decision by applications if they were taking any. So we cant fix it like this

Reply:
Made changes to return `UnsafeRow`. Created new case class `UnionCommandExec  `which overrides `LeafExecNode` and handles the conversion.